File: nomad-samples/js/account-updates/account/change-email.js
import * as prjConstants from "../constants/project-constants.js";
import apiExceptionHandler from "../exceptions/api-exception-handler.js";

export default async function changeEmail(AUTH_TOKEN) 
{
    // Create header for the request
    const HEADERS = new Headers();
    HEADERS.append("Content-Type", "application/json");
    HEADERS.append("Authorization", `Bearer ${AUTH_TOKEN}`);
  
  	const BODY = {
        password: "currentPassword",
        newEmail: "newEmail@newEmail.net"
    };
  
  	// Post
    const RESPONSE = await fetch(`${prjConstants.PORTAL_API_URL}/account/change-email`, {
        method: "POST",
        headers: HEADERS,
        body: JSON.stringify(BODY)
    }).catch((exception) => {
        throw exception;
    });

    // Check for success
    if (RESPONSE.ok) {
      return(RESPONSE.status);
    }
		
  	// There was an error
    apiExceptionHandler(RESPONSE, "Change email failed");
}

File: nomad-samples/portal/py/account_updates/account/change_email.py
from constants.project_constants import *
from exceptions.api_exception_handler import *

import json, requests

def change_email(AUTH_TOKEN: str, EMAIL: str, PASSWORD: str) -> dict:
    if (not AUTH_TOKEN):
        raise Exception("Authentication Token: The authentication token is invalid")
  
    API_URL = PORTAL_API_URL + "/account/change-email"
  
    # Create header for the request
    HEADERS = {
        "Content-Type": "application/json",
        "Authorization": "Bearer " + AUTH_TOKEN
    }

    # Build the payload body
    BODY = {
        "password": PASSWORD,
        "newEmail": EMAIL
    }
    
    try:
        RESPONSE = requests.post(API_URL, headers=HEADERS, data=json.dumps(BODY))
        if not RESPONSE.ok:
            raise Exception()

        return json.loads(RESPONSE.text)
    except:
        api_exception_handler(RESPONSE, "Change email failed")